### PR TITLE
oop-classes-and-objects / it-academy / JC1-117-22 / Ekaterina Komleva / Done all tasks

### PR DESCRIPTION
**Pull request by KateKoml**
## Pull request state

Actual status for PR:
- [x] implemented
- [x] reviewed
- [x] fixed
- [x] approved

### CODE REVIEW

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Time.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+public class Time {
+    private int hours;
+    private int minutes;
+    private int seconds;
+
+    {
+        this.hours = 0;
+        this.minutes = 0;
+        this.seconds = 0;
+    }
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Time.java:12]: избыточно

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Time.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    Time() {
+    }
+
+    Time(int totalSeconds) {
+        final int secondsInMin = 60;
+        final int secondsInHour = 3600;
+        final int secondsInDay = 86400;
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Time.java:20]: "Violation: 'NAME_CONSTANT' was not used for constant naming
Solution: Rename constant"

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Time.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        return seconds;
+    }
+
+    public void setSeconds(int seconds) {
+        this.seconds = seconds;
+    }
+
+    public int getTotalSeconds() {
+        final int secondsInMin = 60;
+        final int secondsInHour = 3600;
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Time.java:58]: "Violation: 'NAME_CONSTANT' was not used for constant naming
Solution: Rename constant"

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Point.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+import com.rakovets.course.java.core.util.NumberUtil;
+
+public class Point {
+    private double x;
+    private double y;
+
+    {
+        this.x = 0;
+        this.y = 0;
+    }
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Point.java:12]: избыточно

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Point.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        this.x = x;
+    }
+
+    public double getY() {
+        return y;
+    }
+
+    public void setY(int y) {
+        this.y = y;
+    }
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Point.java:36]: "Violation: Getters and setters should be placed at the very end
Solution: Move getters and setters to the end"

File: src/test/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/TimeTests.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
+package com.rakovets.course.java.core.practice.oop_classes_and_objects;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class TimeTests {
+
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/TimeTests.java:7]: "Violation: Useless blank lines
Solution: Remove blank lines"

File: src/test/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/TestPoint.java [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+public class TestPoint {
+
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/TestPoint.java:12]: "Violation: Useless blank lines
Solution: Remove blank lines"

File: src/test/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/TimeTests.java [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+    @Test
+    public void testGetTimeGetTotalSeconds() {
+        Time time = new Time();
+        Assertions.assertEquals(0, time.getHours());
+        Assertions.assertEquals(0, time.getMinutes());
+        Assertions.assertEquals(0, time.getSeconds());
+        Assertions.assertEquals(0, time.getTotalSeconds());
+    }
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/TimeTests.java:15]: делаем тесты атомарными

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Rectangle.java [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+        this.topLeftPoint = topLeftPoint;
+    }
+
+    public Point getBottomRightPoint() {
+        return bottomRightPoint;
+    }
+
+    public void setBottomRightPoint(Point bottomRightPoint) {
+        this.bottomRightPoint = bottomRightPoint;
+    }
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Rectangle.java:26]: "Violation: Getters and setters should be placed at the very end
Solution: Move getters and setters to the end"

File: src/test/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/TestRectangle.java [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideArgumentsForGetPerimeter")
+    void testGetPerimeter(Point topLeftPoint, Point bottomRightPoint, double expected) {
+        Rectangle rectangle = new Rectangle(topLeftPoint, bottomRightPoint);
+        double actual = rectangle.getPerimeter(topLeftPoint, bottomRightPoint);
+        Assertions.assertEquals(expected, actual);
+    }
+    @ParameterizedTest
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/TestRectangle.java:33]:
```suggestion
    }
    
    @ParameterizedTest
```

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Atm.java [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+    private int numberBanknotes100;
+
+    {
+        this.numberBanknotes20 = 0;
+        this.numberBanknotes50 = 0;
+        this.numberBanknotes100 = 0;
+    }
+
+    Atm() {
+    }
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Atm.java:15]: избыточно

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Atm.java [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+                numberBanknotes50++;
+            }
+            while (amount > 0 && numberBanknotes20 > 0) {
+                amount -= 20;
+                numberBanknotes20--;
+            }
+            isPossible = true;
+        }
+        return isPossible;
+    }
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Atm.java:79]: не думаю что это сработает

**Review by rakovets — APPROVED** [review]

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Atm.java (hunk 12 above)
Comment (KateKoml) [reply src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Atm.java:79]: Тесты правильно прошли. На каких еще примерах можно проверить?

**Pull request closed without merging**